src/main/java/ch/jalu/configme/resource/yaml/SnakeYamlNodeContainerImpl.java

```diff
@@ -0,0 +1,85 @@
+package ch.jalu.configme.resource.yaml;
+
+import org.jetbrains.annotations.NotNull;
+import org.yaml.snakeyaml.DumperOptions;
+import org.yaml.snakeyaml.comments.CommentLine;
+import org.yaml.snakeyaml.nodes.MappingNode;
+import org.yaml.snakeyaml.nodes.Node;
+import org.yaml.snakeyaml.nodes.NodeTuple;
+import org.yaml.snakeyaml.nodes.Tag;
+
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+/**
+ * Implementation of {@link SnakeYamlNodeContainer}.
+ */
+public class SnakeYamlNodeContainerImpl implements SnakeYamlNodeContainer {
+
+    private final List<String> comments;
```

Comment:
* Rename to rootComments
* Is it OK that they're ignored if we only have one root value?